### DISCUSSION TITLE
Details about client on the server behind istio proxy

### DISCUSSION THREAD

**Discussion by ruz** [istio/istio#52639] [Q&A]
Hi, in certain situations on the HTTP server in the mesh I want to know as much as possible about the client. Trying to figure out way to do it.

I really like what is in x-envoy-peer-metadata header. However, proxy removes this header and doesn't pass it further. Is it possible "preserve" the header? May be with EnvoyFilter record and a custom code?

Can I use some API request and Hash from x-forwarded-client-cert header to get more info?

**Comment by ramaraochavali** [comment]
x-forwarded-client-cert is the header you should rely on to identify and know more about clients

**Comment by ruz** [comment]
x-forwarded-client-cert is not nearly enough. For example:

```
x-forwarded-client-cert: By=spiffe://cluster.local/ns/res/sa/default;Hash=1218696f9fb1ba9f7a3ba3fd8e4bb40d8b6a8e6aa41fba826f170fa4687bf2e7;Subject="";URI=spiffe://cluster.local/ns/res/sa/default
```

This request came from other k8s cluster via eastwest GW.

If I send request from "local" cluster then the only difference would be the Hash field.

Is there API that I can use to get more info by Hash? If not then this header is pretty much useless.
